src/here-studio.ts

```diff
@@ -0,0 +1,342 @@
+
+async function showProject (id : any) {
+    const open = require("open");
+    open(
+        studioBaseURL+"/viewer/?project_id="+id
+        , { wait: false });
+}
+
+async function deleteProject  (id : any, options: any) {
+    console.log("Deleting project : "+id)
+
+    //If project exists send a DELETE request for that projectID
+    const uri = "/project-api/projects/"+id;
+    const cType = "";
+    let response = await execute (uri, "DELETE", cType, "", options.token);
+
+    if (response && response.statusCode === 204) {
+        console.log("Successfully deleted project.")
+    }
+    else {
+        console.log("Unable to delete project having project-id: "+id)
+    }
+}
+
+
+/**
+ * Will fetch all projects
+ * @param options
+ */
+async function findAllProjects (options:any) {
+    try {
+        const uri = "/project-api/projects";
+        const cType = "";
+        let response = await execute(uri, "GET", cType, "", options.token);
+        return response;
+    } catch (error) {
+        console.log("Unable to get all project data")
+        return null;
+    }
+}
+
+/**
+ * Will list all the projects for the given user in below format
+ *
+ * @param options
+ */
+export async function listProjects (options: any) {
+    console.log("Please wait; Fetching your list of projects...")
+
+    let response = await findAllProjects(options)
+    let body = JSON.parse(response.body);
+    if (response.body.length == 0) {
+        console.log("No xyz projects found");
+    } else {
+        let fields = ["id", "title", "status"];
+
+        //Flattened array of project JsonObjects containing info about name, id and description, add any other info later as necessary
+        let extractProjectInfo: any[] = new Array();
+
+        //Iterate through all the projects and extract meta information in extractColumns Array having JSON Objects with keys of id, name and description,
+        body.map((currentProject:any) => {
+
+            //Check whether meta info like project description and name exists for that project? - > If exists Push the meta info with id in new
+            if (currentProject.hasOwnProperty("meta")) {
+                let viewerURL = "";
+                if (currentProject.status.toUpperCase() === "PUBLISHED") {
+                    viewerURL = "https://xyz.here.com/viewer/?project_id=" + currentProject.id;
+                }
+                let currentProjectDetails = {
+                    id: currentProject.id,
+                    title: currentProject.meta.name,
+                    description: currentProject.meta.description,
+                    status: currentProject.status,
+                    viewerURL
+                }
+                extractProjectInfo.push(new Object(currentProjectDetails))
+            }
+        })
+
+        //List the project
+        common.drawNewTable(extractProjectInfo, fields, [40, 25, 12]);
+    }
+}
+
+common.validate(commands, [process.argv[2]], program);
+prompter.stop();
```

Comment:
do we need prompter  as we are not asking any input from the user?

Reply:
Have removed prompter in the input